quinn-proto/src/connection/send_buffer.rs

```diff
@@ -76,22 +76,41 @@ impl SendBuffer {
 
     /// Compute the next range to transmit on this stream and update state to account for that
     /// transmission
-    pub fn poll_transmit(&mut self, max_len: usize) -> Range<u64> {
+    pub fn poll_transmit(&mut self, mut max_len: usize) -> (Range<u64>, bool) {
```

Comment:
Before I fix/update the unit-tests, there's the question of whether people prefer this approach (assuming  `max_len` only covers the pure data size, and adjust it), or to pass the `max_data` here which doesn't includes the full remaining size after a stream ID would have been encoded (16 bytes more), and let this function do the remaining calculations.

I would prefer the latter.

Reply:
The latter does seem more attractive, although I find it a little hard to visualize.

Reply:
To clarify, you're proposing refactoring such that the value passed in is the amount of space available to be used for both the length tag and the actual data, but already accounting for the space used by the stream ID? That sounds good to me.

Reply:
Exactly. I will update for this. Will be a tiny change